src/main/java/be/lghs/accounting/configuration/SecurityUserService.java

```diff
@@ -11,28 +13,49 @@
 
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class SecurityUserService extends DefaultOAuth2UserService {
 
     private final UserRepository userRepository;
 
+    private static final Logger logger = LoggerFactory.getLogger(SecurityUserService.class);
+
+    String getMandatory(Map<String, Object> values, String attribute){
+        if(values.containsKey(attribute)){
+            return (String) values.get(attribute);
+        } else {
+            logger.error("Missing attribute " + attribute + " for user " +
+                    values.entrySet().stream().map(e-> e.getKey() +":"+e.getValue())
+                            .collect(Collectors.joining(";")));
+
+            throw new RuntimeException("Missing attribute " + attribute + " for user ");
+        }
```

Comment:
```suggestion
        var value = (String) values.get(attribute);
        if (value == null) {
            logger.error("Missing attribute {} for user {}", attribute, values);
            throw new RuntimeException("Missing attribute " + attribute + " for user");
        }
        return value;
```

Reply:
I removed the log altogether because there already is a log after.  
not particularly a fan of extracting a variable and checking directly for null but if you really want I can do it.